brian2/equations/equations.py

```diff
@@ -433,6 +437,25 @@ def __init__(self, type, varname, dimensions, var_type=FLOAT, expr=None,
                                                       if variable =='xi' or variable.startswith('xi_')]),
                                     doc='Stochastic variables in the RHS of this equation')
 
+
+    _state_tuple = property(lambda self: (self.type, self.varname,
+                                          self.dim, self.var_type,
+                                          self.expr, tuple(self.flags)),
+                            doc='A tuple representing the full state of this '
+                                'object, used for hashing and equality '
+                                'testing.')
```

Comment:
As above - it's not about making it clearer it's about protecting ourselves from our future selves. I think this is particularly important for this cacheing stuff because it would be so easy to make a mistake here that we have to go to some efforts to avoid doing so.